tuf/api/metadata.py

```diff
@@ -1497,25 +1665,78 @@ def from_dict(cls, delegations_dict: Dict[str, Any]) -> "Delegations":
         keys_res = {}
         for keyid, key_dict in keys.items():
             keys_res[keyid] = Key.from_dict(keyid, key_dict)
-        roles = delegations_dict.pop("roles")
-        roles_res: Dict[str, DelegatedRole] = {}
-        for role_dict in roles:
-            new_role = DelegatedRole.from_dict(role_dict)
-            if new_role.name in roles_res:
-                raise ValueError(f"Duplicate role {new_role.name}")
-            roles_res[new_role.name] = new_role
+        roles = delegations_dict.pop("roles", None)
+        roles_res: Optional[Dict[str, DelegatedRole]] = None
+
+        if roles is not None:
+            roles_res = {}
+            for role_dict in roles:
+                new_role = DelegatedRole.from_dict(role_dict)
+                if new_role.name in roles_res:
+                    raise ValueError(f"Duplicate role {new_role.name}")
+                roles_res[new_role.name] = new_role
+
+        succinct_roles_dict = delegations_dict.pop("succinct_roles", None)
+        succinct_roles_info = None
+        if succinct_roles_dict is not None:
+            succinct_roles_info = SuccinctRoles.from_dict(succinct_roles_dict)
+
         # All fields left in the delegations_dict are unrecognized.
-        return cls(keys_res, roles_res, delegations_dict)
+        return cls(keys_res, roles_res, succinct_roles_info, delegations_dict)
 
     def to_dict(self) -> Dict[str, Any]:
         """Returns the dict representation of self."""
         keys = {keyid: key.to_dict() for keyid, key in self.keys.items()}
-        roles = [role_obj.to_dict() for role_obj in self.roles.values()]
-        return {
+        res_dict: Dict[str, Any] = {
             "keys": keys,
-            "roles": roles,
             **self.unrecognized_fields,
         }
+        if isinstance(self.roles, dict):
+            roles = [role_obj.to_dict() for role_obj in self.roles.values()]
+            res_dict["roles"] = roles
+        elif isinstance(self.succinct_roles, SuccinctRoles):
+            res_dict["succinct_roles"] = self.succinct_roles.to_dict()
+        return res_dict
+
+    def find_delegation(self, target_filepath: str) -> Optional[str]:
```

Comment:
Currently, It's not used in the code, but my reasoning was that I thought that having a method that gives you the correct delegation now when we support succinct roles and standard roles seemed a must-have.
Now, when I think about it we have `get_all_delegations` which does actually the same thing, but instead of returning just one role, it returns all roles responsible for that `target_filepath` which could be many in the case of standard roles.

You are right, we have a replacement for this method, and given it's not currently used it's better to remove it.
Again, we can always add it later if the need arises.